provider/spark.go

```diff
@@ -626,8 +630,172 @@ func (e EMRExecutor) InitializeExecutor(store FileStore) error {
 	return store.Write(sparkScriptPath, buff)
 }
 
-func NewSparkExecutor(execType SparkExecutorType, config SparkExecutorConfig, logger *zap.SugaredLogger) (SparkExecutor, error) {
+type SparkGenericConfig struct {
+	Master        string
+	DeployMode    string
+	PythonVersion string
+}
+
+func (sc *SparkGenericConfig) Deserialize(config SerializedConfig) error {
+	err := json.Unmarshal(config, sc)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (sc *SparkGenericConfig) Serialize() ([]byte, error) {
+	conf, err := json.Marshal(sc)
+	if err != nil {
+		return nil, err
+	}
+	return conf, nil
+}
+
+func (sc *SparkGenericConfig) IsExecutorConfig() bool {
+	return true
+}
+
+type SparkGenericExecutor struct {
+	master        string
+	deployMode    string
+	pythonVersion string
+	logger        *zap.SugaredLogger
+}
+
+func (s *SparkGenericExecutor) InitializeExecutor(store FileStore) error {
+	s.logger.Info("Uploading PySpark script to filestore")
+	sparkScriptPath := helpers.GetEnv("SPARK_SCRIPT_PATH", "/scripts/offline_store_spark_runner.py")
+	sparkScriptPathWithPrefix := store.PathWithPrefix(sparkScriptPath, false)
+
+	err := readAndUploadFile(sparkScriptPath, sparkScriptPathWithPrefix, store)
+	scriptExists, _ := store.Exists(sparkScriptPathWithPrefix)
+	if err != nil && !scriptExists {
+		return fmt.Errorf("could not upload spark script: Path: %s, Error: %v", sparkScriptPathWithPrefix, err)
+	}
+	return nil
+}
+
+func (s *SparkGenericExecutor) RunSparkJob(args *[]string, store FileStore) error {
```

Comment:
Why is this a pointer to a slice?

Reply:
No particular reason except that the interface was designed this way. I can change it so it accepts `[]string`

Reply:
A slice pointer is super weird, I'd change it if its easy to do so

Reply:
Can also leave it to a separate PR for next release if prefered let me know

Reply:
so databricks executor uses it and I believe that's why we made into a pointer. we will be working on databricks next so we will address it in that PR.